decryptor/mysql/proxy_test.go

```diff
@@ -0,0 +1,55 @@
+package mysql
+
+import (
+	"context"
+	"github.com/cossacklabs/acra/decryptor/base"
+	"github.com/cossacklabs/acra/encryptor/config"
+	"testing"
+)
+
+type decryptorFactory struct{}
+
+func (*decryptorFactory) New(clientID []byte) (base.Decryptor, error) {
+	return nil, nil
+}
+
+type tableSchemaStore struct{empty bool}
+
+func (*tableSchemaStore) GetTableSchema(tableName string) *config.TableSchema {
+	panic("implement me")
+}
+
+func (store *tableSchemaStore) IsEmpty() bool {
+	return store.empty
+}
+
+func TestEncryptorTurnOnOff(t *testing.T) {
+	emptyStore := &tableSchemaStore{true}
+	nonEmptyStore := &tableSchemaStore{false}
+	setting := base.NewProxySetting(&decryptorFactory{}, emptyStore, nil, nil, nil)
+	proxyFactory, err := NewProxyFactory(setting)
+	if err != nil {
+		t.Fatal(setting)
+	}
+	proxy, err := proxyFactory.New(context.TODO(), nil, nil, nil)
```

Comment:
Why did you create new proxy factory with `context.TODO()` in mysql and with nil in postgres?

Reply:
It turns out that you need sometimes to pass some context when you create proxy in mysql case, but not in postgres right?

Reply:
yes. mysql proxy implementation initialize logger in constructor - https://github.com/cossacklabs/acra/blob/master/decryptor/mysql/response_proxy.go#L175
but postgresql just save context (https://github.com/cossacklabs/acra/blob/master/decryptor/postgresql/pg_decryptor.go#L109) and create logger after (https://github.com/cossacklabs/acra/blob/master/decryptor/postgresql/pg_decryptor.go#L127, https://github.com/cossacklabs/acra/blob/master/decryptor/postgresql/pg_decryptor.go#L481)